receiver/k8sclusterreceiver/factory.go

```diff
@@ -37,17 +37,15 @@ const (
 
 var defaultNodeConditionsToReport = []string{"Ready"}
 
-var _ component.ReceiverFactoryOld = (*Factory)(nil)
-
-// Factory is the factory for kubernetes-cluster receiver.
-type Factory struct {
-}
-
-func (f Factory) Type() configmodels.Type {
-	return typeStr
+type receiverOptions struct {
```

Comment:
Maybe a different PR, but confused why not using config directly and have this temp options between receiver and config.